protocol/x/subaccounts/keeper/transfer.go

```diff
@@ -250,17 +250,28 @@ func (k Keeper) DistributeFees(
 			return err
 		}
 
-		// emit a metric for the amount of fees transferred to the market mapper
-		if revShare.RevShareType == revsharetypes.REV_SHARE_TYPE_MARKET_MAPPER {
-			labels := []metrics.Label{
-				metrics.GetLabelForIntValue(metrics.MarketId, int(perpetual.Params.MarketId)),
-			}
-			metrics.AddSampleWithLabels(
-				metrics.MarketMapperRevenueDistribution,
-				metrics.GetMetricValueFromBigInt(revShare.QuoteQuantums),
-				labels...,
-			)
+		// emit a metric for revenue share
+		labels := []metrics.Label{}
+		var metricName string
+		switch revShare.RevShareType {
+		case revsharetypes.REV_SHARE_TYPE_MARKET_MAPPER:
```

Comment:
This switching casing code seems redundant, why not just emit a generic rev share variable with the rev share type + recipient as labels? 

If you need a separate metrics for `market_mapper`, then conditionally add a `marketId` label if ` revShare.RevShareType == revsharetypes.REV_SHARE_TYPE_MARKET_MAPPER`

Reply:
Wanted to keep existing metric as is for marketmapper

Will update the other metrics though!